server/index.js

```diff
@@ -24,102 +24,132 @@ const portfinder = require('portfinder');
 const crypto = require("crypto");
 const fse = require('fs-extra');
 
+const DEFAULT_PORT = 9000;
 
-async function start(portPreferred) {
-    let port = process.env.ASSET_COMPUTE_DEV_PORT || '9000';
-    if (!isNaN(portPreferred)) {
-        port = portPreferred;
+class DevtoolServer {
+
+    /**
+     * Start server and open developer tool UI in a browser
+     * @param {*} preferredPort preferred port to bind server, can be a number or string
+     * 
+     * preferredPort will take precedence over the default port and the environment variable port
+     * however, if the requested port is taken, it will find and usethe closest open port
+     */
+    async run (preferredPort) {
+        this.port = process.env.ASSET_COMPUTE_DEV_PORT || DEFAULT_PORT;
+        if (!isNaN(preferredPort)) {
+            this.port = preferredPort;
+        }
+    
+        const portRangeServer = {
+            port: this.port, // minimum port
+            stopPort: 9999 // maximum port
+        };
+        this.port = await findOpenPort(portRangeServer);
+
+        const randomString = getRandomString();
+    
+        
+    
+        // Get port from environment and store in Express.
+        app.set('port', this.port);
+        app.set('devToolToken', randomString);
+    
+        // Create HTTP server.
+        this.server = http.createServer(app);
+    
+        this.server.listen(this.port);
+        this.server.on('error', error => {
+            if (error.syscall !== 'listen') {
+                throw new Error(error);
+            }
+
+            // handle specific listen errors with friendly messages
+            const bind = typeof this.port === 'string'
+                ? 'Pipe ' + this.port
+                : 'Port ' + this.port;
+            switch (error.code) {
+            case 'EACCES':
+                console.error(bind,  'requires elevated privileges');
+                break;
+            case 'EADDRINUSE':
+                console.error(bind, ' is already in use');
+                break;
+            }
+            throw new Error(error);
+        });
+
+        this.server.on('listening', () => onListening(this.server, randomString));
+        this.server.on('close', () => {
+            console.log("Asset Compute Developer Tool Server Stopped");
+        });
     }
 
-    const portRangeServer = {
-        port: port, // minimum port
-        stopPort: 9999 // maximum port
-    };
-    port = await findOpenPort(portRangeServer);
+    async stop() {
+        return this.server.close();
+    }
+
+}
+
+// for backwards compatibility
```

Comment:
`aio-cli-plugin-asset-compute` should be the only consumers of the devtool library, right? I don't think 3rd party folks use this.

In this case we don't have to keep backwards compatibility. Just drop `start()`, and release with `BREAKING-RELEASE:`.

But it's up to you...

Reply:
well its also a little easier for the start scripts in the server to keep the `start` function: https://github.com/adobe/asset-compute-devtool/blob/master/server/package.json#L19